h/api/search/test/client_test.py

```diff
@@ -10,56 +10,47 @@
 @pytest.mark.usefixtures('elasticsearch', 'presenters')
 class TestIndexAnnotation:
 
-    def test_it_presents_the_annotation(self):
+    def test_it_presents_the_annotation(self, client_obj, presenters):
         request = mock.Mock()
         annotation = mock.Mock()
 
-        self.client().index_annotation(request, annotation)
+        client_obj.index_annotation(request, annotation)
 
-        client.presenters.AnnotationJSONPresenter.assert_called_once_with(
+        presenters.AnnotationJSONPresenter.assert_called_once_with(
             request, annotation)
 
-    def test_it_indexes_the_annotation(self):
-        client_ = self.client()
+    def test_it_indexes_the_annotation(self, client_obj, presenters):
+        client_obj.index_annotation(mock.Mock(), mock.Mock())
 
-        client_.index_annotation(mock.Mock(), mock.Mock())
-
-        client_.conn.index.assert_called_once_with(
+        client_obj.conn.index.assert_called_once_with(
             index='hypothesis',
             doc_type='annotation',
-            body=client.presenters.AnnotationJSONPresenter.return_value.asdict.return_value,
+            body=presenters.AnnotationJSONPresenter.return_value.asdict.return_value,
             id='test_annotation_id',
         )
 
-    def test_it_inserts_the_source_field(self):
-        client_ = self.client()
+    def test_it_inserts_the_source_field(self, client_obj):
         annotation = mock.Mock()
 
-        client_.index_annotation(mock.Mock(), annotation)
+        client_obj.index_annotation(mock.Mock(), annotation)
 
-        assert client_.conn.index.call_args[1]['body']['target'][0]['scope'] == [
-            annotation.target_uri_normalized]
+        assert client_obj.conn.index.call_args[1]['body']['target'][0]['scope'] == [annotation.target_uri_normalized]
 
-    def client(self):
+    @pytest.fixture
+    def client_obj(self):
```

Comment:
But what's stopping you from importing at the top of the test module?

I'm merely suggesting that you separate the fixture definition from the module namespace. What you're proposing is entirely compatible with a fixture defined in a single class which looks like:

```python
@pytest.fixture
def client():
    from h.api.search.client import Client
    return Client(...)
```

And the `foobar` function would undoubtedly not be tested in the same class as the `Client` class. I'm not seeing where there's a conflict...